setInterval/3-setIntervalPolyfill.js

```diff
@@ -0,0 +1,72 @@
+const {
+    setIntervalPolyfill,
+    clearIntervalPolyfill
+}= (()=>{
+
+    // Closure Scope.
+    var intervalId = 0;
+    var intervalMap = {}
+
+    function setIntervalPolyfill (callbackFn, delay=0) {
+
+        // Throw error if callback is not a function.
+        if(typeof callbackFn !== "function"){
+            throw new TypeError('"callback" argument must be a funtion.')
+        }
+
+        var id = intervalId++;
+        intervalMap[id] = true;
+
+        // to be passed as paremeters in callback function call.
+        var args = Array.prototype.slice.call(arguments,2);
+
+        (function repeat(){
+            setTimeout(
+                ()=>{
+                    if(intervalMap[id]){
+                        callbackFn(args);
+                        repeat();
+                    }  
+                },
+                delay
+            );
+        })();
```

Comment:
Why are we creating IIFEs?

Reply:
I wanted to call the function for the first time.
Should I call in the next line for better understanding?

Reply:
It is okay if you are doing so but it introduces additional layer of information that is `IIFE` so either we need to explain that in the video or just go for the simpler way.